### PR TITLE
Relpace `LoopOp` Op with `DoLoopOp` Op in FIR Dialect

### PR DESCRIPTION
**Pull request by SouraVX**

### CODE REVIEW

File: flang/documentation/BurnsideToFIR.md [hunk 1]
```diff
@@ -428,7 +428,7 @@ Fortran into FIR in a semantics preserving way.
     // create a temporary to hold E(V)
     %v = ... : !fir.array<4:i32>
     %8 = fir.alloca !fir.array<4:f32> : !fir.ref<!fir.array<4:f32>>
-    fir.loop %i = %c1 to %c4 unordered {
+    fir.doloop %i = %c1 to %c4 unordered {
```
Comment (schweitzpgi) [inline flang/documentation/BurnsideToFIR.md:431]: Should be `do_loop` in the syntactic form like this. Although the `_` is omitted in the camelcase class name.

File: flang/include/flang/Optimizer/Transforms/Passes.td [hunk 2]
```diff
@@ -17,7 +17,7 @@
 include "mlir/Pass/PassBase.td"
 
 def AffineDialectPromotion : FunctionPass<"promote-to-affine"> {
-  let summary = "Promotes fir.loop and fir.where to affine.for and affine.if where possible";
+  let summary = "Promotes fir.doloop and fir.where to affine.for and affine.if where possible";
```
Comment (schweitzpgi) [inline flang/include/flang/Optimizer/Transforms/Passes.td:20]: 1 more here

File: flang/lib/Lower/Bridge.cpp [hunk 3]
```diff
@@ -813,11 +813,11 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     assert(info.stepValue && "step value must be set");
     info.loopVariable = createTemp(loc, *info.loopVariableSym);
 
-    // Structured loop - generate fir.loop.
+    // Structured loop - generate fir.doloop.
```
Comment (schweitzpgi) [inline flang/lib/Lower/Bridge.cpp:816]: and here

File: flang/lib/Lower/Bridge.cpp [hunk 4]
```diff
@@ -864,7 +864,7 @@ class FirConverter : public Fortran::lower::AbstractConverter {
   void genFIRIncrementLoopEnd(IncrementLoopInfo &info) {
     auto loc = toLocation();
     if (info.isStructured()) {
-      // End fir.loop.
+      // End fir.doloop.
```
Comment (schweitzpgi) [inline flang/lib/Lower/Bridge.cpp:867]: also here

File: flang/lib/Optimizer/Transforms/AffinePromotion.cpp [hunk 5]
```diff
@@ -362,17 +362,17 @@ void rewriteMemoryOps(Block *block, mlir::PatternRewriter &rewriter) {
   }
 }
 
-/// Convert `fir.loop` to `affine.for`, creates fir.convert for arrays to
+/// Convert `fir.doloop` to `affine.for`, creates fir.convert for arrays to
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Transforms/AffinePromotion.cpp:365]: arg

File: flang/lib/Optimizer/Transforms/AffinePromotion.cpp [hunk 6]
```diff
@@ -496,7 +496,7 @@ class AffineIfConversion : public mlir::OpRewritePattern<fir::IfOp> {
   AffineFunctionAnalysis &functionAnalysis;
 };
 
-/// Promote fir.loop and fir.if to affine.for and affine.if, in the cases
+/// Promote fir.doloop and fir.if to affine.for and affine.if, in the cases
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Transforms/AffinePromotion.cpp:499]: sigh

**Comment by SouraVX** [comment]
Done!

**Pull request merged**